include/or_ompl/StateSpaces.h

```diff
@@ -94,7 +99,11 @@ class ContinuousJointsProjectionEvaluator : public ompl::base::ProjectionEvaluat
         ompl::base::ProjectionMatrix _projectionMatrix;
 };
 
+#ifdef OR_OMPL_HAS_BOOSTSMARTPTRS
 typedef boost::shared_ptr<ContinuousJointsProjectionEvaluator> ContinuousJointsProjectionEvaluatorPtr;
+#else
+typedef std::shared_ptr<ContinuousJointsProjectionEvaluator> ContinuousJointsProjectionEvaluatorPtr;
+#endif
```

Comment:
Please combine these into a single `#ifdef` block, ideally in a dedicated `.h` file so we don't clutter up the code with scattered definitions.